src/platform/packages/shared/kbn-test/src/jest/transforms/babel/index.test.js

```diff
@@ -0,0 +1,168 @@
+    expect(k3).not.toBe(k1);
+
+    // Same inputs produce the same key
+    const k1b = transformer.getCacheKey('console.log("a")', __filename, opts);
+    expect(k1b).toBe(k1);
+  });
+});
+
+// Re-implemented integration test verifying that the same file transformed under
+// two Jest configs (differing only in a non-cache-key option) only runs the
+// transformer once by leveraging the shared cache directory.
+
+describe('jest CLI cache deduplication', () => {
+  it('reuses transformed output across two configs', () => {
+    // Ensure the data dir exists
+    fs.mkdirSync(path.join(REPO_ROOT, 'data'), { recursive: true });
+
+    // Use the same directory the transformer uses
+    const cacheDir = path.join(REPO_ROOT, 'data', 'jest-cache');
+
+    // Create a temp dir to hold the test files
+    const tmpDir = fs.mkdtempSync(path.join(REPO_ROOT, 'data', 'jest-dedupe-'));
+
+    // Test module that requires transformation
+    const srcFile = path.join(tmpDir, 'sample.ts');
+
+    // Test module that requires transformation
+    fs.writeFileSync(srcFile, 'export const VALUE:number = 42;', 'utf8');
+
+    // Create a Jest test that imports the module
+    const testFile = path.join(tmpDir, 'sample.test.ts');
+
+    // Create a Jest test that imports the module
+    fs.writeFileSync(
+      testFile,
+      `import { VALUE } from './sample';
+      
+      it('should validate if a module can be imported', () => {
+        expect(VALUE).toBe(42); 
+      });
+      `,
+      'utf8'
+    );
+
+    // File that will hold a counter of how many times the transformer was invoked
+    const counterFile = path.join(tmpDir, 'transform-count.txt');
+
+    fs.writeFileSync(counterFile, '0', 'utf8');
+
+    if (fs.existsSync(cacheDir)) {
+      for (const e of fs.readdirSync(cacheDir)) {
+        fs.rmSync(path.join(cacheDir, e), { recursive: true, force: true });
+      }
+    }
+
+    // Wrapper transformer increments a counter only when transforming the source module (not the test file)
+    const realTransformerPath = path.join(
+      REPO_ROOT,
+      'src/platform/packages/shared/kbn-test/src/jest/transforms/babel/index.js'
+    );
+    const wrapperPath = path.join(tmpDir, 'wrapper-transformer.js');
+
+    // Create a custom transformer that wraps the real one and increments a counter
+    // we only want to count transformations of the source file, not the test file
+    fs.writeFileSync(
+      wrapperPath,
+      `
+const fs = require('fs');
+const real = require(${JSON.stringify(realTransformerPath)});
+const counterFile = ${JSON.stringify(counterFile)};
+module.exports = { 
+  ...real,
+  process(src, filename, ...rest) {  
+    if (filename && filename.endsWith('sample.ts')) {
+      try { 
+        const c = parseInt(fs.readFileSync(counterFile, 'utf8') ,10) || 0;
+
+        fs.writeFileSync(counterFile, String(c+1));
+      } catch(_){
+      }
+    }
+  return real.process(src, filename, ...rest);
+}};`,
+      'utf8'
+    );
```

Comment:
You can also consider reading the cache directory after running both configs and counting the files that start with `sample_logs`. I think it's more straight-forward because you don't have to wrap the transformer and you're validating the actual outcome rather than monitoring the process. Also it'd be nice to have a test that validates scenarios in which the cache is not reused, like a different NODE_ENV, BABEL_ENV or rootDir, using the same testing approach, so we can have more confidence in the test itself.